crates/webidl/src/util.rs

```diff
@@ -34,35 +38,380 @@ pub fn mdn_doc(class: &str, method: Option<&str>) -> String {
+            Int8Array(s) => s.to_syn_type(record, pos),
+            Int16Array(s) => s.to_syn_type(record, pos),
+            Int32Array(s) => s.to_syn_type(record, pos),
+            Uint8Array(s) => s.to_syn_type(record, pos),
+            Uint8ClampedArray(s) => s.to_syn_type(record, pos),
+            Uint16Array(s) => s.to_syn_type(record, pos),
+            Uint32Array(s) => s.to_syn_type(record, pos),
+
+            DOMString(s) => s.to_syn_type(record, pos),
+            ByteString(s) => s.to_syn_type(record, pos),
+            USVString(s) => s.to_syn_type(record, pos),
+
+            // Support for these types is not yet implemented, so skip
+            // generating any bindings for this function.
+            ArrayBuffer(_)
+            | DataView(_)
+            | Error(_)
+            | FrozenArrayType(_)
+            | Object(_)
+            | Promise(_)
+            | RecordType(..)
+            | Sequence(_)
+            | Symbol(_) => {
+                None
+            }
+        }
     }
 }
 
+
 /// Map a webidl const value to the correct wasm-bindgen const value
-pub fn webidl_const_v_to_backend_const_v(v: &webidl::ast::ConstValue) -> backend::ast::ConstValue {
+pub fn webidl_const_v_to_backend_const_v(v: &ConstValue) -> backend::ast::ConstValue {
+    use std::f64::{NEG_INFINITY, INFINITY, NAN};
+    use backend::ast;
+
     match *v {
-        webidl::ast::ConstValue::BooleanLiteral(b) => backend::ast::ConstValue::BooleanLiteral(b),
-        webidl::ast::ConstValue::FloatLiteral(f) => backend::ast::ConstValue::FloatLiteral(f),
-        webidl::ast::ConstValue::SignedIntegerLiteral(i) => backend::ast::ConstValue::SignedIntegerLiteral(i),
-        webidl::ast::ConstValue::UnsignedIntegerLiteral(u) => backend::ast::ConstValue::UnsignedIntegerLiteral(u),
-        webidl::ast::ConstValue::Null => backend::ast::ConstValue::Null,
+        ConstValue::Boolean(b) => ast::ConstValue::BooleanLiteral(b.0),
+        ConstValue::Float(FloatLit::NegInfinity(_)) => {
+            ast::ConstValue::FloatLiteral(NEG_INFINITY)
+        }
+        ConstValue::Float(FloatLit::Infinity(_)) => {
+            ast::ConstValue::FloatLiteral(INFINITY)
+        }
+        ConstValue::Float(FloatLit::NaN(_)) => {
+            ast::ConstValue::FloatLiteral(NAN)
+        }
+        ConstValue::Float(FloatLit::Value(s)) => {
+            ast::ConstValue::FloatLiteral(s.0.parse().unwrap())
+        }
+        ConstValue::Integer(lit) => {
+            let mklit = |orig_text: &str, base: u32, offset: usize| {
+                let (negative, text) = if orig_text.starts_with("-") {
+                    (true, &orig_text[1..])
+                } else {
+                    (false, orig_text)
+                };
+                if text == "0" {
+                    return ast::ConstValue::SignedIntegerLiteral(0)
+                }
+                let text = &text[offset..];
+                match i64::from_str_radix(text, base) {
```

Comment:
Why can't we just parse it as a u64 in the first place? I.e. something like this:
```rust
let n = u64::from_str_radix(text, base).unwrap_or_else(|_| panic!("literal too big: {}", orig_text));
if negative {
    let n = if n > (i64::min_value() as u64).wrapping_neg() {
        panic!("literal too big: {}", orig_text)
    } else {
        n.wrapping_neg() as i64
    };
    ast::ConstValue::SignedIntegerLiteral(n)
} else {
    ast::ConstValue::UnsignedIntegerLiteral(n)
}
```

Reply:
Ah that's much better!